### PR TITLE
Refactor pod runner

### PR DESCRIPTION
**Pull request by e-sumin**
## Change Overview

Minor refactoring of PodRunner, reuse functionality of PodController under the hood.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/pod_command_executor_test.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+var _ = Suite(&PodCommandExecutorTestSuite{})
+
+const (
+	podCommandExecutorNS            = "pod-runner-test"
+	podCommandExecutorPodName       = "test-pod"
+	podCommandExecutorContainerName = "test-container"
+)
+
+func (s *PodCommandExecutorTestSuite) SetUpSuite(c *C) {
+	os.Setenv("POD_NAMESPACE", podCommandExecutorNS)
```
Comment (denisvmedia) [inline pkg/kube/pod_command_executor_test.go:41]: I think, we should reset it once we are done with the tests.

File: pkg/kube/pod_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+var _ = Suite(&PodControllerTestSuite{})
+
+const (
+	podControllerNS      = "pod-runner-test"
+	podControllerPodName = "test-pod"
+)
+
+func (s *PodControllerTestSuite) SetUpSuite(c *C) {
+	os.Setenv("POD_NAMESPACE", podControllerNS)
```
Comment (denisvmedia) [inline pkg/kube/pod_controller_test.go:41]: Ditto.

File: pkg/kube/pod_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	. "gopkg.in/check.v1"
+	"k8s.io/api/core/v1"
```
Comment (denisvmedia) [inline pkg/kube/pod_controller_test.go:25]: I'd alias it to `corev1`.

**Review by denisvmedia — APPROVED** [review]
LGTM (with few minor issues, which aren't show stoppers)

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**